pandas/core/frame.py

```diff
@@ -3195,6 +3195,11 @@ def __setitem__(self, key, value):
             self._setitem_array(key, value)
         elif isinstance(value, DataFrame):
             self._set_item_frame_value(key, value)
+        elif is_list_like(value) and 1 < len(
+            self.columns.get_indexer_for([key])
+        ) == len(value):
```

Comment:
not necessarily for this PR, but it might be worth implementing a `Index.count` method (mirroring `list.count`)?

Reply:
Yeah this would be helpful probably, something like that would have helped in #39278